### PR TITLE
Catch case when windows do not span entire sequence for pair coalescence stats

### PR DESCRIPTION
**Pull request by nspope**
Fixes issue reported here: https://github.com/tskit-dev/tskit/issues/2986#issuecomment-2361134470

### CODE REVIEW

**Comment by nspope** [comment]
The issue was not passing `TSK_REQUIRE_FULL_SPAN` to `tsk_treeseq_check_windows` -- and I noticed that `tsk_treeseq_divergence_matrix` also doesn't do this here: https://github.com/tskit-dev/tskit/blob/dcee409933ce826dc1d672d4dc0b267e0bcc2d4b/c/tskit/trees.c#L8463

which may be a bug?

Also I wonder if it'd be better to move this option to the header rather than defining inline?

https://github.com/tskit-dev/tskit/blob/dcee409933ce826dc1d672d4dc0b267e0bcc2d4b/c/tskit/trees.c#L1194

**Comment by nspope** [comment]
This is ready @jeromekelleher

File: python/tests/test_lowlevel.py [hunk 1]
```diff
@@ -4402,8 +4402,8 @@ def test_c_tsk_err_node_out_of_bounds(self, bad_node):
     def test_c_tsk_err_bad_windows(self):
         ts = self.example_ts()
         L = ts.get_sequence_length()
-        with pytest.raises(_tskit.LibraryError, match="BAD_WINDOWS"):
-            self.pair_coalescence_counts(ts, windows=[-1.0, L])
+        with pytest.raises(tskit.LibraryError, match="TSK_ERR_BAD_WINDOWS"):
```
Comment (petrelharp) [inline python/tests/test_lowlevel.py:4405]: should all these be `_tskit.LibraryError`?

Comment (nspope) [reply python/tests/test_lowlevel.py:4405]: what's the distinction?

Comment (jeromekelleher) [reply python/tests/test_lowlevel.py:4405]: I don't think it matters, it's probably the same exception. The _tskit module is the low-level C Python module, which defines LibraryError. Somewhere in Python we're importing this exception from _tskit, which then becomes tskit.LibraryError also.

So I guess Peter's right, it's better to directly catch the exception that we know the C code will throw.

**Review by jeromekelleher — COMMENTED** [review]
Probably better to be consistent with other code on where the exception is coming from all right.

File: python/tests/test_lowlevel.py (hunk 1 above)
Comment (nspope) [reply python/tests/test_lowlevel.py:4405]: shall I go ahead and make this change across the board in `tests/test_lowlevel.py`, then? there's a number of places (outside of the pair coalescence stuff) where `tskit.LibraryError` is checked for

**Comment by nspope** [comment]
Done-- there's a number of other places in test_lowlevel.py where `tskit.LibraryError` is checked for rather than `_tskit.LibraryError`, but I left those alone here.

**Review by jeromekelleher — APPROVED** [review]

**Pull request merged**